responders/src/API/EMBC.Responders.API/Controllers/ProfileController.cs

```diff
@@ -78,14 +94,21 @@ public async Task<ActionResult<UserProfile>> GetCurrentUserProfile()
         /// <returns>profile id</returns>
         [HttpPost("current")]
         [ProducesResponseType(StatusCodes.Status200OK)]
-        public async Task<ActionResult> Update(UpdateUserProfileRequest request)
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<IActionResult> Update(UpdateUserProfileRequest request)
         {
-            var memberId = User.FindFirstValue(ClaimTypes.Sid);
-            var teamId = User.FindFirstValue("user_team");
+            var userName = User.FindFirstValue(ClaimTypes.Upn).Split('@')[0];
+            var userId = User.FindFirstValue(ClaimTypes.Sid);
+            var sourceSystem = User.FindFirstValue("identity_source");
 
             // Get the current user
-            var reply = await messagingClient.Send(new TeamMembersQueryCommand { TeamId = teamId, MemberId = memberId, IncludeActiveUsersOnly = false });
+            var reply = await messagingClient.Send(new TeamMembersQueryCommand { UserName = userName, IncludeActiveUsersOnly = false });
```

Comment:
It is fine to use member id - the user already authenticated so it has the external user id set correctly